theano/sandbox/gpuarray/basic_ops.py

```diff
@@ -599,50 +681,53 @@ def c_code(self, node, name, inputs, outputs, sub):
     def c_code_cache_version(self):
         return (3,)
 
-
 cuda_from_gpu = CudaFromGpu()
 
 
 class GpuAlloc(HideC, Alloc):
-    def __init__(self, memset_0=False):
+    __props__ = ('memset_0', 'context') + Alloc.__props__
```

Comment:
I both agree and disagree with your comment.

I agree because you are right that we loose all properties from Alloc since we don't call its \__init__ method.

I disagree because I had too many problems with op appearing a equal while being not really equal because of some properties of the superclass.  This both ensure that we take Alloc's props into account and that Alloc does use the \__props__ mechanism so that we won't have its \__eq__ shadow ours.

This could be fixed by a tweaking of the \__props__ implementation to enable it to override parent methods.